hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessageReader.java

```diff
@@ -65,6 +74,23 @@ private boolean readFrame(ByteBuffer src) {
         }
         if (readOffset == -1) {
             int frameLength = Bits.readIntL(src, src.position());
+            if (!trusted) {
+                if (frameLength > maxFrameLength) {
+                    throw new IllegalArgumentException(
```

Comment:
why not use `MaxMessageSizeExceeded`?

Reply:
Changed to suggested `MaxMessageSizeExceeded`.